test/bricks-net-api.test.js

```diff
@@ -0,0 +1,634 @@
+							assert.equal(requestBody, receivedBody);
+							
+							// Test reading non-chunked response.
+							response.writeHead(200, {
+								'Content-Type': requestContentType,
+								'Content-Length': requestBody.length
+							});
+							response.end(requestBody);
+						}
+						catch (ex) {
+							done(ex);
+						}
+					});
+				}
+				catch (ex) {
+					done(ex);
+				}
+			};
+			
+			var response = api.HTTP(
+				api.POST("localhost:20000/test", requestBody, requestContentType)
+			);
+			
+			assert.equal(true, response instanceof api.HTTPResponse, 'response instanceof HTTPResponse');
+			assert.equal(true, when.isPromiseLike(response.body), 'response.body isPromiseLike');
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body, '`body` is a string');
+				assert.equal(requestBody, body, '`body` equals to the response body');
+				done();
+			}).done(undefined, done);
+		});
+		
+		/* TODO(sompylasar): Investigate further on this test failure.
+		it('should make a `POST` request with a large body (' + largeBodyLength + ' bytes) non-chunked', function (done) {
+			serverRequestHandler = function (request, response) {
+				try {
+					assert.equal('POST', request.method);
+					assert.equal('/test', request.url);
+					assert.equal('1.1', request.httpVersion);
+					
+					// WARNING: `request.rawHeaders` requires Node.js 0.12.0
+					assert.deepEqual([
+						'Host',
+						'localhost',
+						'Content-Type',
+						'text/plain',
+						'Content-Length',
+						largeBodyLength
+					], request.rawHeaders);
+					
+					var body = '';
+					request.on('data', function (chunk) {
+						body += chunk.toString();
+					});
+					request.on('end', function () {
+						try {
+							// Test sending large string request.
+							assert.equal(largeBody, body);
+							
+							// Test receiving large non-chunked response.
+							response.writeHead(200, {
+								'Content-Type': 'application/json',
+								'Content-Length': requestJson.length
+							});
+							response.end(largeBody);
+						}
+						catch (ex) {
+							done(ex);
+						}
+					});
+				}
+				catch (ex) {
+					done(ex);
+				}
+			};
+			
+			var response = api.HTTP(
+				api.POST("localhost:20000/test", largeBody, "text/plain")
+			);
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body, '`body` is a string');
+				assert.equal(largeBody, body, '`body` equals to the response body');
+				done();
+			}).done(undefined, done);
+		});
+		// */
+		
+		it('should make a `POST` request with a large body (' + largeBodyLength + ' bytes) chunked', function (done) {
+			serverRequestHandler = function (request, response) {
+				try {
+					assert.equal('POST', request.method);
+					assert.equal('/test', request.url);
+					assert.equal('1.1', request.httpVersion);
+					
+					// WARNING: `request.rawHeaders` requires Node.js 0.12.0
+					assert.deepEqual([
+						'Host',
```

Comment:
It's not that specific. I tend to use double quotes in objects to mimic JSON notation which does not allow single quotes (or not use quotes at all). Otherwise, I use single quotes because they are easier to type. The double quotes in this code base are often either a copy-paste from C++ or another code base or a swap for convenience if I need to put single quotes inside. There are some tools to highlight these inconsistencies, maybe we could run some if absolutely required.